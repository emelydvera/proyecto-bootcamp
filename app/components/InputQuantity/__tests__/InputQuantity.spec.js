const React = require('react');
const { screen, render, fireEvent } = require('@testing-library/react');
const InputQuantity = require('..');

describe('InputQuantity component', () => {
    let component;
    let error = '';
    let setError = jest.fn(value => { error = value; });
    let setQuantityToBuy = jest.fn();
    let input;

    beforeEach(() => {
        component = render(<InputQuantity
            className={'checkout__input'}
            tabIndex={12}
            error={error}
            setError={setError}
            quantityToBuy={'3'}
            setQuantityToBuy={setQuantityToBuy}
            availableQuantity={2}
        />);
        input = screen.getByRole('spinbutton', {
            name: /ingrese la cantidad de productos a comprar/i
        });
    });
    it('should render correctly', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    it('when entering a value in the input the setQuantityToBuy function is called with the entered value', () => {
        fireEvent.change(input, { target: { value: '2' } });
        expect(error).toBe('');
        expect(setQuantityToBuy).toHaveBeenCalledWith('2');
    });

    it('if the value entered in input is negative , setQuantityToBuy is called with "1" as value', () => {
        fireEvent.change(input, { target: { value: '-1' } });
        expect(error).toBe('');
        expect(setQuantityToBuy).toHaveBeenCalledWith('1');
    });

    it('if the value entered in the input is greater than the quantity available, an error message appears indicating the quantity available to buy', () => {
        fireEvent.change(input, { target: { value: '4' } });
        expect(setError).toHaveBeenCalledWith('Puedes comprar hasta 2 unidades');
        expect(setQuantityToBuy).toHaveBeenCalledWith('4');
        expect(error).toBe('Puedes comprar hasta 2 unidades');
    });

    it('if the value of the input is empty, setError must be called with "Puedes comprar mínimo 1 unidad" being this the error message', () => {
        fireEvent.change(input, { target: { value: '' } });
        expect(setError).toHaveBeenCalledWith('Puedes comprar mínimo 1 unidad');
        expect(setQuantityToBuy).toHaveBeenCalledWith('');
        expect(error).toBe('Puedes comprar mínimo 1 unidad');
    });
});