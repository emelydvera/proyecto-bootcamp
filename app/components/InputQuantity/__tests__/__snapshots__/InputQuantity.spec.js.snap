// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputQuantity component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="input__quantity"
  >
    <div
      class="andes-form-control andes-form-control--textfield andes-form-control--centered andes-form-control--default checkout__input"
    >
      <label>
        <span
          class="andes-form-control__label"
        >
          Cantidad
        </span>
        <div
          class="andes-form-control__control"
        >
          <input
            aria-label="Ingrese la cantidad de productos a comprar"
            class="andes-form-control__field"
            maxlength="120"
            rows="1"
            tabindex="12"
            type="number"
            value="3"
          />
        </div>
      </label>
      <div
        class="andes-form-control__bottom"
      >
        <span
          class="andes-form-control__message"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
